src/main/java/org/jabref/gui/referencemetadata/ReferenceMetadataFetcherAction.java

```diff
@@ -0,0 +1,126 @@
+package org.jabref.gui.referencemetadata;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import javax.swing.undo.UndoManager;
+
+import javafx.collections.ObservableList;
+
+import org.jabref.gui.DialogService;
+import org.jabref.gui.JabRefFrame;
+import org.jabref.gui.StateManager;
+import org.jabref.gui.actions.SimpleCommand;
+import org.jabref.gui.undo.NamedCompound;
+import org.jabref.gui.util.BindingsHelper;
+import org.jabref.gui.util.TaskExecutor;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.preferences.JabRefPreferences;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.jabref.gui.actions.ActionHelper.needsDatabase;
+import static org.jabref.gui.actions.ActionHelper.needsEntriesSelected;
+
+/**
+ * This class allows fetching the reference metadata (e.g. citation counts) from the web for the currently selected
+ * entries in a library.
+ */
+public class ReferenceMetadataFetcherAction extends SimpleCommand {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ReferenceMetadataFetcherAction.class);
+
+    private static boolean USE_REFERENCE_METADATA_FETCHER_GOOGLE_SCHOLAR = true;
+    private static boolean USE_REFERENCE_METADATA_FETCHER_SEMANTIC_SCHOLAR = true;
+    private static boolean USE_REFERENCE_METADATA_FETCHER_OPEN_CITATIONS = true;
+
+    private final DialogService dialogService;
+    private final JabRefPreferences preferences;
+    private final StateManager stateManager;
+    private UndoManager undoManager;
+    private TaskExecutor taskExecutor;
+
+    public ReferenceMetadataFetcherAction(JabRefFrame frame, JabRefPreferences preferences, StateManager stateManager, UndoManager undoManager, TaskExecutor taskExecutor) {
+        this.dialogService = frame.getDialogService();
+        this.preferences = preferences;
+        this.stateManager = stateManager;
+        this.undoManager = undoManager;
+        this.taskExecutor = taskExecutor;
+
+        this.executable.bind(needsDatabase(this.stateManager).and(needsEntriesSelected(stateManager)));
+        this.statusMessage.bind(BindingsHelper.ifThenElse(executable, Localization.lang("This operation fetches " +
+                "reference metadata for the currently selected entries online."), Localization.lang("This " +
+                "operation requires one or more entries to be selected.")));
```

Comment:
I think the localization test is failing because of this statement. Please concatenate the strings inside the Localization:lang-expressions two single strings and add them to the l10n-resources.

Reply:
Only to the english resource file. Translations are handled by crowdin.